ui-react/packages/atlasmap-ui/stories/Mapper.stories.tsx

```diff
@@ -28,6 +29,7 @@ export const sample = () => createElement(() => {
       targets={targets}
       mappings={mappings}
       addToMapping={addToMapping}
+      importAtlasFile={action('importAtlasFile')}
```

Comment:
Can you encapsulate these function pointers into a class? This ends up with having all the function pointers required for all downstream components and everytime this file have to be edited. Please decouple function pointers from unrelated components.

Reply:
This is a story file for Storybook, it’s specific to the ui package and its development workflow. It has no impacts whatsoever on upstream.

Reply:
Ah sorry I put it on a wrong file, I meant real components that has this issue

Reply:
Right, I’m not sure I’m following. What’s the problem?

Reply:
e.g. this one, seems like every components has same problem, the number of parameters will eventually exponentially explode
https://github.com/atlasmap/atlasmap/blob/master/ui-react/packages/atlasmap-ui/src/mapper/Mapper.tsx#L22-L28

Reply:
That's the `Mapper`'s interface, it's the data and callbacks that component needs to work in a self-contained way.

Reply:
It's not a problem, it's what's needed to achieve a decoupling of the UI elements from the underlying backend. This will allow us to implement things as the ones @chirino was suggesting in the chat a while ago.

Reply:
I don't get it why it's not a problem, why we have to carry all function pointers  without any wrapping all the way down to the terminal components? It's fatally worse than carrying ConfigModel.

Reply:
Function pointers? It's a Typescript interface, it's the type definition of the properties required by the component to work. It's a compile time thing.

Reply:
I mean individual functions are more internal implementation details than ConfigModel as an interface to the AtlasMap main body (i.e. ui/models and ui/services)

Reply:
There's really two parts to this - there's the complexity of the atlasmap-provider which I'll address by splitting things out in a way similar to what we have in atlasmap-data-mapper/components.  Then there's the complexity of the UI <-> atlasmap-provider interface which is a reflection of the added support components.  Can I suggest that for now we keep the interface simple and add to it all of the functions that are required.  It should be obvious after it's all there and we can consider how to establish new classes to organize the interface (likely along the lines established by atlasmap-provider).  We can keep cfg as an interface object from atlasmap-provider to the systems and models packages.